ipi/engine/motion/constrained_dynamics.py

```diff
@@ -0,0 +1,498 @@
+"""Contains the classes that deal with the different dynamics required in
+different types of ensembles.
+
+Holds the algorithms required for normal mode propagators, and the objects to
+do the constant temperature and pressure algorithms. Also calculates the
+appropriate conserved energy quantity for the ensemble of choice.
+"""
```

Comment:
May be nice to update these as well.

Reply:
Especially because the constraints have not been implemented with constant pressure dynamics. Do you think that bit can be quickly added ? Anyway I will return tomorrow morning.